### ISSUE TITLE
Document OneHotArray

### ISSUE THREAD

**Issue by cossio** [FluxML/Flux.jl#1519]
It seems that the new `OneHotArray` (https://github.com/FluxML/Flux.jl/pull/1448) is not documented. There is nothing mentioned in https://fluxml.ai/Flux.jl/dev/data/onehot/ and it's not clear how one should use `OneHotArray`. Can we add some documentation? @darsnack

**Comment by darsnack** [comment]
Yup the docs for the old `onehotbatch` etc. should be there, but I'll add more docs around the new type.
